roles/ceph-config/templates/ceph.conf.j2

```diff
@@ -82,18 +82,14 @@ filestore xattr use omap = true
 {% if osd_objectstore == 'bluestore' %}
 {% set _num_osds = num_osds | default(0) | int %}
 [osd]
-{% if is_hci | bool and _num_osds > 0 %}
+{% if is_hci and _num_osds > 0 and not osd_memory_target is defined %}
 {# hci_safety_factor is the safety factor for HCI deployments #}
-{% if ansible_memtotal_mb * 1048576 * hci_safety_factor / _num_osds > osd_memory_target %}
 {% set _osd_memory_target = (ansible_memtotal_mb * 1048576 * hci_safety_factor / _num_osds) | int %}
-{% endif %}
-{% elif _num_osds > 0 %}
+{% elif _num_osds > 0 and not osd_memory_target is defined %}
 {# non_hci_safety_factor is the safety factor for dedicated nodes #}
-{% if ansible_memtotal_mb * 1048576 * non_hci_safety_factor / _num_osds > osd_memory_target %}
 {% set _osd_memory_target = (ansible_memtotal_mb * 1048576 * non_hci_safety_factor / _num_osds) | int %}
 {% endif %}
-{% endif %}
-osd memory target = {{ _osd_memory_target | default(osd_memory_target) }}
+osd memory target = {{ _osd_memory_target | default(osd_memory_target | default(0)) }}
```

Comment:
If you use default(0) then you will have `osd memory target = 0` when there's no OSDs which breaks the backward compatibility.

Reply:
Yes I can see how this is bad in this edge case. I fixed this by not including any `osd memory target` value into the file if neither `_osd_memory_target` and `osd_memory_target` are set.